third_party/terraform/utils/validation.go

```diff
@@ -261,3 +261,16 @@ func StringNotInSlice(invalid []string, ignoreCase bool) schema.SchemaValidateFu
 		return
 	}
 }
+
+// Ensure that hourly timestamp strings "HH:MM" have the minutes zeroed out for hourly only inputs
+func validateHourlyOnly(val interface{}, key string) (warns []string, errs []error) {
+	v := val.(string)
+	parts := strings.Split(v, ":")
+	if len(parts) != 2 {
```

Comment:
Feel free to ignore this but we also might as well verify the hour is between 0 and 23 at this point? Part of me thinks you should make this a regexp match or do time.Parse("15:04", v) (even though I hate how they make you define a custom format)

Reply:
we don't _have_ to validate hours because the api will reject bad inputs there whereas the minutes it would accept and silently transform. But it's probably worth throwing it in anyway. 

re: time.parse or regex - it's going to be a significantly more complex operation to use either of those over string splitting and casting to ints and I don't think either strategy provides added value in readability or flexibility.